nfqueue/nfqueue.go

```diff
@@ -459,33 +459,37 @@ func (q *NfQueue) Recv() (*common.NfqGenMsg, []*common.NfAttrResponsePayload, er
 }
 
 //ProcessPackets -- Function to wait on socket to receive packets and post it back to channel
-func (q *NfQueue) ProcessPackets() {
+func (q *NfQueue) ProcessPackets(ctx context.Context) {
 	for {
-		nfgenmsg, attr, err := q.Recv()
-
-		if err != nil {
-			if q.errorCallback != nil {
-				q.errorCallback(fmt.Errorf("Netlink error %v", err), nfgenmsg)
-				continue
-			} else {
-				fmt.Println("Received Error from netlink", err)
-				if nfgenmsg != nil {
-					fmt.Println(nfgenmsg.GetNfgenFamily())
+		select {
+		case <-ctx.Done():
+			q.NfqClose()
+			return
+		default:
+			nfgenmsg, attr, err := q.Recv()
```

Comment:
This is a blocking call so if we are waiting here for packets the ctx.Done will never be sampled